descqa/AGN_NumberDensities.py

```diff
@@ -0,0 +1,347 @@
+from __future__ import unicode_literals, absolute_import, division
+import os
+import re
+import numpy as np
+from scipy.interpolate import interp1d
+from GCR import GCRQuery
+try:
+    from itertools import zip_longest
+except ImportError:
+    from itertools import izip_longest as zip_longest
+from .plotting import plt
+from .base import BaseValidationTest, TestResult
+
+possible_observations = {
+    'SDSS': {
+        'filename_template': 'AGNdata/SDSS/richards_2006_table2.dat',
+        'usecols': {'g': {'dN/dm':(0, 1, 2, 5),
+                          'N<m':(0, 3, 4, 5),
+                         },
+                    'i': {'dN/dm':(0, 6, 7, 10),
+                          'N<m':(0, 8, 9, 10),
+                         },
+                   },
+        'colnames': ('mag', 'n', 'err', 'N_Q'),
+        'skiprows': 5,
+        'missing_values':'...',
+        'bin-width': 0.25,
+        'label': 'SDSS DR3',
+        'color': 'r',
+        'title': 'Richards et. al. (2006)  $\\rm M_{}<{}$, $\\rm {}<z<{}$',
+        'ytitle':{'dN/dm':'$\\rm N (deg^{{-2}} {}mag^{{-1}})$',
+                  'N<m':'$\\rm N(<{}) (deg^{{-2}})$',
+                 },
+    },
+}
+
+__all__ = ['AGN_NumberDensity']
+
+
+class AGN_NumberDensity(BaseValidationTest):
+    """
+    AGN number desnsity test
+    """
+    def __init__(self, band='g', rest_frame_band='i', Mag_lim=-22.5, z_lim=[0.4, 2.1],
```

Comment:
Avoid using a list as argument default.
```suggestion
    def __init__(self, band='g', rest_frame_band='i', Mag_lim=-22.5, z_lim=(0.4, 2.1),
```

Reply:
@jiwoncpark Here is the pull request. Thanks for doing the scientific review.